File: elixir/3179.ex
defmodule Solution do
  @spec value_after_k_seconds(n :: integer, k :: integer) :: integer
  def value_after_k_seconds(n, k) do
    step(k+1, n, 0, 1) |> rem(1_000_000_007)
  end

  def step(_k, n, cur_n, acc) when n-1 == cur_n, do: acc
  def step(k, n, cur_n, acc) do
    new_acc = k*acc |> div(cur_n+1)
    step(k+1, n, cur_n+1, new_acc)
  end
end

File: python/3179.py
class Solution:
    def valueAfterKSeconds(self, n: int, k: int) -> int:
        return self.step(k+1, n, 0, 1)

    def step(self, k, n, cur_n, acc):
        if n-1 == cur_n: return acc % 1_000_000_007
        new_acc = k*acc // (cur_n+1) 
        return self.step(k+1, n, cur_n+1, new_acc)